core/trino-spi/src/test/java/io/trino/spi/block/BenchmarkVariableWidthBlock.java

```diff
@@ -0,0 +1,197 @@
+
+        private int[] positionsIds;
+        private Block variableWidthBlock;
+
+        @Setup
+        public void setup()
+        {
+            positionsIds = selectedPositions.generateIds(selectedPositionsCount);
+            Slice[] slices = createExpectedValues();
+            if (nullsAllowed) {
+                slices = alternatingNullValues(slices);
+            }
+            variableWidthBlock = createBlockBuilderWithValues(slices).build();
+        }
+
+        private static Slice[] createExpectedValues()
+        {
+            Slice[] expectedValues = new Slice[POSITIONS];
+            for (int position = 0; position < POSITIONS; position++) {
+                expectedValues[position] = createExpectedValue(position);
+            }
+            return expectedValues;
+        }
+
+        private static Slice createExpectedValue(int length)
+        {
+            DynamicSliceOutput dynamicSliceOutput = new DynamicSliceOutput(16);
+            for (int index = 0; index < length; index++) {
+                dynamicSliceOutput.writeByte(length * (index + 1));
+            }
+            return dynamicSliceOutput.slice();
+        }
+
+        private static <T> T[] alternatingNullValues(T[] objects)
+        {
+            T[] objectsWithNulls = Arrays.copyOf(objects, objects.length * 2 + 1);
+            for (int i = 0; i < objects.length; i++) {
+                objectsWithNulls[i * 2] = null;
+                objectsWithNulls[i * 2 + 1] = objects[i];
+            }
+            objectsWithNulls[objectsWithNulls.length - 1] = null;
+            return objectsWithNulls;
+        }
+
+        private static BlockBuilder createBlockBuilderWithValues(Slice[] expectedValues)
+        {
+            BlockBuilder blockBuilder = new VariableWidthBlockBuilder(null, expectedValues.length, 32 * expectedValues.length);
+            for (Slice expectedValue : expectedValues) {
+                if (expectedValue == null) {
+                    blockBuilder.appendNull();
+                }
+                else {
+                    blockBuilder.writeBytes(expectedValue, 0, expectedValue.length()).closeEntry();
+                }
+            }
+            return blockBuilder;
+        }
+
+        public int[] getPositionsIds()
+        {
+            return positionsIds;
+        }
+
+        public Block getVariableWidthBlock()
+        {
+            return variableWidthBlock;
+        }
+    }
+
+    public enum SelectedPositions
+    {
+        ALL {
+            @Override
+            int[] generateIds(int positionsCount)
+            {
+                return IntStream.range(0, positionsCount).toArray();
+            }
+        },
+        RANDOM {
+            @Override
+            int[] generateIds(int positionsCount)
+            {
+                return new Random(42).ints(positionsCount, 0, POSITIONS).toArray();
+            }
+        },
+        GROUPED {
```

Comment:
it's not very different from SEQUENCE in perf numbers but it is a more realistic scenario in actual queries as a full sequence would usually result in call to `getRegion` rather than `copyPositions`